pkg/watcher/reconciler/dynamic/dynamic_test.go

```diff
@@ -251,7 +252,7 @@ func TestReconcile_PipelineRun(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	r := NewDynamicReconciler(results, prclient, nil)
+	r := NewDynamicReconciler(resultsClient, logsClient, prclient, nil)
```

Comment:
Are we doing any testing with the logsClient in these tests to see if the log file was actually written?

Reply:
The API handler for this has a unit test, but from reconciling -> API handler -> logs written doesn't have a test. As of now we cannot add this kind of test S3 backend, but test for File storage backend can be implemented.

Reply:
@adambkaplan @dibyom 
The problem in implementing this in the reconciler unit test is that we are using `tkncli` to collect logs, which is getting it from the `pod` logs. But in these unit tests we are creating only `PipelineRuns` and `TaskRuns` but no pods.
We have covered writing of the logs in the `UpdateLog` handler of the API server, and in this case we can use an e2e test, which will be easier.